aten/src/ATen/core/function_schema.h

```diff
@@ -147,6 +188,49 @@ struct FunctionSchema {
             is_vararg,
             is_varret) {}
 
+  // check whether this schema is backward compatible with old_schema,
+  // which means all the existing call sites of old_schema can execute with
+  // this schema. we intent to be conservative, and if necessary, we may
+  // relax the checks in future. the following conditions are considered as
+  // this schema is backward compatible with old_schema:
+  //   1) two schemas are equal
+  //   2) two schemas have same number of arguments, and this schema's
+  //      arguments are backward compatible with the corresponding ones in
+  //      argument list of old_schema.
+  //   3) this schema has m argument, old_argument has n argument, m > n.
+  //      the first n arguments of this schema are backward compatible with
+  //      the corresponding arguments of old_schema. the remaning arguments
+  //      must be either OptionalType or provide default values.
+  bool isBackwardCompatibleWith(const FunctionSchema& old_schema) const {
+    if (*this == old_schema) {
+      return true;
+    }
+    if (!(name() == old_schema.name()
+      && overload_name() == old_schema.overload_name()
+      && is_vararg() == old_schema.is_vararg()
+      && is_varret() == old_schema.is_varret()
+      && returns() == old_schema.returns()
```

Comment:
This is using Argument ==, is that was is intended rather than Arugment backward compatible?

Reply:
originally, i just intended to be conservative. Now I changed the logical, use a inversed normal argument backcompat check.